nixos/modules/services/networking/godns.nix

```
@@ -0,0 +1,87 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    types
+    ;
+
+  cfg = config.services.godns;
+
+  configFormat = pkgs.formats.yaml { };
+  configFile =
+    if cfg.configFile == null then configFormat.generate "config.yaml" cfg.config else cfg.configFile;
+in
+{
+
+  options.services.godns = {
+    enable = mkEnableOption "GoDNS service";
+
+    package = mkPackageOption pkgs "godns" { };
+
+    config = mkOption {
```

For consistency with other modules.
```suggestion
    settings = mkOption {
```